crml/sylo/src/device.rs

```diff
@@ -13,49 +13,122 @@
 *     https://centrality.ai/licenses/lgplv3.txt
 */
 
-use frame_support::{decl_event, decl_module, decl_storage, dispatch::DispatchResult, dispatch::Vec, ensure};
-use frame_system;
+use frame_support::{decl_error, decl_module, decl_storage, dispatch::DispatchResult, dispatch::Vec, ensure};
 
 const MAX_DEVICES: usize = 1000;
 
-pub trait Trait: frame_system::Trait {
-	/// The overarching event type.
-	type Event: From<Event<Self>> + Into<<Self as frame_system::Trait>::Event>;
-}
+type DeviceId = u32;
+
+pub trait Trait: frame_system::Trait {}
 
 decl_module! {
 	pub struct Module<T: Trait> for enum Call where origin: T::Origin, system = frame_system {
-		fn deposit_event() = default;
+		type Error = Error<T>;
 	}
 }
 
 // The data that is stored
 decl_storage! {
 	trait Store for Module<T: Trait> as SyloDevice {
-		pub Devices get(devices): map hasher(blake2_128_concat) T::AccountId => Vec<u32>;
+		pub Devices get(devices): map hasher(blake2_128_concat) T::AccountId => Vec<DeviceId>;
 	}
 }
 
-decl_event!(
-	pub enum Event<T> where <T as frame_system::Trait>::Hash, <T as frame_system::Trait>::AccountId {
-		DeviceAdded(AccountId, Hash, u32),
+decl_error! {
+	pub enum Error for Module<T: Trait> {
+		/// There are no devices registered for user (missing user_id in Devices)
+		UserIdMissing,
+		/// Device is already registered to user (device_id is already in use)
+		DeviceId,
```

Comment:
Should make these self descriptive since they generate strings for the client e.g. `DeviceIdExists`

Reply:
Sorry mentioned this multiple times lol